### ISSUE TITLE
FATAL ]ExpertDispatcher::GPUExecFunc DenseActDense tensor_ids: [4 5 6 3 9 10 11 8 14 15 16 13 ] expert_type 4 mat1 and mat2 shapes cannot be multiplied (3x4096 and 14336x1)

### ISSUE THREAD

**Issue by kungfup** [EfficientMoE/MoE-Infinity#42]
/root/fcc/software/conda/envs/moe-infinity/lib/python3.10/site-packages/auto_gptq/nn_modules/triton_utils/kernels.py:411: FutureWarning: `torch.cuda.amp.custom_fwd(args...)` is deprecated. Please use `torch.amp.custom_fwd(args..., device_type='cuda')` instead.
  def forward(ctx, input, qweight, scales, qzeros, g_idx, bits, maxq):
/root/fcc/software/conda/envs/moe-infinity/lib/python3.10/site-packages/auto_gptq/nn_modules/triton_utils/kernels.py:419: FutureWarning: `torch.cuda.amp.custom_bwd(args...)` is deprecated. Please use `torch.amp.custom_bwd(args..., device_type='cuda')` instead.
  def backward(ctx, grad_output):
/root/fcc/software/conda/envs/moe-infinity/lib/python3.10/site-packages/auto_gptq/nn_modules/triton_utils/kernels.py:461: FutureWarning: `torch.cuda.amp.custom_fwd(args...)` is deprecated. Please use `torch.amp.custom_fwd(args..., device_type='cuda')` instead.
  @custom_fwd(cast_inputs=torch.float16)
CUDA extension not installed.
CUDA extension not installed.
Do not detect pre-installed ops, use JIT mode
tokenizer_config.json: 967B [00:00, 2.75MB/s]                                                         
tokenizer.model: 100%|██████████████████████████████████████████████| 493k/493k [00:02<00:00, 175kB/s]
tokenizer.json: 1.80MB [00:04, 446kB/s]
special_tokens_map.json: 100%|██████████████████████████████████████| 72.0/72.0 [00:00<00:00, 294kB/s]
config.json: 2.21kB [00:00, 6.14MB/s]                                                                 
generation_config.json: 100%|█████████████████████████████████████████| 116/116 [00:00<00:00, 458kB/s]
quantize_config.json: 100%|███████████████████████████████████████████| 185/185 [00:00<00:00, 746kB/s]
README.md: 22.0kB [00:00, 421kB/s]                                          | 0.00/185 [00:00<?, ?B/s]
.gitattributes: 1.52kB [00:00, 6.47MB/s]                                                              
model.safetensors: 100%|█████████████████████████████████████████| 23.8G/23.8G [36:55<00:00, 10.7MB/s]
Fetching 10 files: 100%|█████████████████████████████████████████████| 10/10 [36:56<00:00, 221.65s/it]
[WARNING] FlashAttention is not available in the current environment. Using default attention.
Using /root/.cache/torch_extensions/py310_cu124 as PyTorch extensions root...
Emitting ninja build file /root/.cache/torch_extensions/py310_cu124/prefetch/build.ninja...
Building extension module prefetch...
Allowing ninja to set a default number of workers... (overridable by setting the environment variable MAX_JOBS=N)
ninja: no work to do.
Loading extension module prefetch...
Time to load prefetch op: 1.1674115657806396 seconds
[20250303 00:10:00.605451Z ][26105 ][INFO  ]Create ArcherAioThread for thread:  0 - archer_aio_thread.cpp:12
[20250303 00:10:00.605625Z ][26105 ][INFO  ]Loading index file from  /data/yzh/moe-infinity/TheBloke/Mixtral-8x7B-v0.1-GPTQ/archer_index - archer_tensor_handle.cpp:44
[20250303 00:10:00.608298Z ][26105 ][INFO  ]Index file size  4579 - archer_tensor_handle.cpp:50
[20250303 00:10:00.608721Z ][26105 ][INFO  ]Device count  4 - archer_prefetch_handle.cpp:40
[20250303 00:10:00.612463Z ][26105 ][INFO  ]Enabled peer access between device  0 1 - archer_prefetch_handle.cpp:56
[20250303 00:10:00.616006Z ][26105 ][INFO  ]Enabled peer access between device  0 2 - archer_prefetch_handle.cpp:56
[20250303 00:10:00.619587Z ][26105 ][INFO  ]Enabled peer access between device  0 3 - archer_prefetch_handle.cpp:56
[20250303 00:10:00.619702Z ][26105 ][INFO  ]Enabled peer access between device  1 0 - archer_prefetch_handle.cpp:56
[20250303 00:10:00.623033Z ][26105 ][INFO  ]Enabled peer access between device  1 2 - archer_prefetch_handle.cpp:56
[20250303 00:10:00.626863Z ][26105 ][INFO  ]Enabled peer access between device  1 3 - archer_prefetch_handle.cpp:56
[20250303 00:10:00.626971Z ][26105 ][INFO  ]Enabled peer access between device  2 0 - archer_prefetch_handle.cpp:56
[20250303 00:10:00.626980Z ][26105 ][INFO  ]Enabled peer access between device  2 1 - archer_prefetch_handle.cpp:56
[20250303 00:10:00.630295Z ][26105 ][INFO  ]Enabled peer access between device  2 3 - archer_prefetch_handle.cpp:56
[20250303 00:10:00.630398Z ][26105 ][INFO  ]Enabled peer access between device  3 0 - archer_prefetch_handle.cpp:56
[20250303 00:10:00.630407Z ][26105 ][INFO  ]Enabled peer access between device  3 1 - archer_prefetch_handle.cpp:56
[20250303 00:10:00.630413Z ][26105 ][INFO  ]Enabled peer access between device  3 2 - archer_prefetch_handle.cpp:56
[20250303 00:10:00.630416Z ][26105 ][INFO  ]Enabled peer access for all devices - archer_prefetch_handle.cpp:63
Loading model from offload_path ...
/root/fcc/software/conda/envs/moe-infinity/lib/python3.10/site-packages/transformers/modeling_utils.py:5006: FutureWarning: `_is_quantized_training_enabled` is going to be deprecated in transformers 4.39.0. Please use `model.hf_quantizer.is_trainable` instead
  warnings.warn(
`loss_type=None` was set in the config but it is unrecognised.Using the default loss: `ForCausalLMLoss`.
MixtralConfig {
  "_name_or_path": "TheBloke/Mixtral-8x7B-v0.1-GPTQ",
  "architectures": [
    "MixtralForCausalLM"
  ],
  "attention_dropout": 0.0,
  "bos_token_id": 1,
  "eos_token_id": 2,
  "hidden_act": "silu",
  "hidden_size": 4096,
  "initializer_range": 0.02,
  "intermediate_size": 14336,
  "max_position_embeddings": 32768,
  "model_type": "mixtral",
  "num_attention_heads": 32,
  "num_experts_per_tok": 2,
  "num_hidden_layers": 32,
  "num_key_value_heads": 8,
  "num_local_experts": 8,
  "output_router_logits": false,
  "pad_token_id": 0,
  "pretraining_tp": 1,
  "quantization_config": {
    "bits": 4,
    "damp_percent": 0.1,
    "desc_act": true,
    "disable_exllama": true,
    "group_size": -1,
    "model_file_base_name": "model",
    "model_name_or_path": null,
    "modules_in_block_to_quantize": [
      [
        "self_attn.k_proj",
        "self_attn.v_proj",
        "self_attn.q_proj"
      ],
      [
        "self_attn.o_proj"
      ],
      [
        "block_sparse_moe.experts.0.w1",
        "block_sparse_moe.experts.0.w2",
        "block_sparse_moe.experts.0.w3"
      ],
      [
        "block_sparse_moe.experts.1.w1",
        "block_sparse_moe.experts.1.w2",
        "block_sparse_moe.experts.1.w3"
      ],
      [
        "block_sparse_moe.experts.2.w1",
        "block_sparse_moe.experts.2.w2",
        "block_sparse_moe.experts.2.w3"
      ],
      [
        "block_sparse_moe.experts.3.w1",
        "block_sparse_moe.experts.3.w2",
        "block_sparse_moe.experts.3.w3"
      ],
      [
        "block_sparse_moe.experts.4.w1",
        "block_sparse_moe.experts.4.w2",
        "block_sparse_moe.experts.4.w3"
      ],
      [
        "block_sparse_moe.experts.5.w1",
        "block_sparse_moe.experts.5.w2",
        "block_sparse_moe.experts.5.w3"
      ],
      [
        "block_sparse_moe.experts.6.w1",
        "block_sparse_moe.experts.6.w2",
        "block_sparse_moe.experts.6.w3"
      ],
      [
        "block_sparse_moe.experts.7.w1",
        "block_sparse_moe.experts.7.w2",
        "block_sparse_moe.experts.7.w3"
      ]
    ],
    "quant_method": "gptq",
    "sym": true,
    "true_sequential": true,
    "use_exllama": false
  },
  "rms_norm_eps": 1e-05,
  "rope_theta": 1000000.0,
  "router_aux_loss_coef": 0.02,
  "router_jitter_noise": 0.0,
  "sliding_window": 4096,
  "tie_word_embeddings": false,
  "torch_dtype": "bfloat16",
  "transformers_version": "4.46.3",
  "use_cache": true,
  "vocab_size": 32000
}

/root/fcc/software/conda/envs/moe-infinity/lib/python3.10/site-packages/transformers/generation/utils.py:1375: UserWarning: Using the model-agnostic default `max_length` (=20) to control the generation length. We recommend setting `max_new_tokens` to control the maximum length of the generation.
  warnings.warn(
/root/fcc/software/conda/envs/moe-infinity/lib/python3.10/site-packages/transformers/generation/utils.py:2097: UserWarning: You are calling .generate() with the `input_ids` being on a device type different than your model's device. `input_ids` is on cuda, whereas the model is on cpu. You may experience unexpected behaviors or slower generation. Please make sure that you have put `input_ids` to the correct device by calling for example input_ids = input_ids.to('cpu') before running `.generate()`.
  warnings.warn(
[20250303 00:10:10.817228Z ][48010 ][FATAL ]ExpertDispatcher::GPUExecFunc DenseActDense tensor_ids: [4 5 6 3 9 10 11 8 14 15 16 13 ] expert_type 4 mat1 and mat2 shapes cannot be multiplied (3x4096 and 14336x1)
Exception raised from meta at /pytorch/aten/src/ATen/native/LinearAlgebra.cpp:201 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::string) + 0x96 (0x7f677736c1b6 in /root/fcc/software/conda/envs/moe-infinity/lib/python3.10/site-packages/torch/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, std::string const&) + 0x64 (0x7f6777315a76 in /root/fcc/software/conda/envs/moe-infinity/lib/python3.10/site-packages/torch/lib/libc10.so)
frame #2: at::meta::structured_mm::meta(at::Tensor const&, at::Tensor const&) + 0x260 (0x7f675ada5aa0 in /root/fcc/software/conda/envs/moe-infinity/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #3: <unknown function> + 0x36ee87c (0x7f67274ee87c in /root/fcc/software/conda/envs/moe-infinity/lib/python3.10/site-packages/torch/lib/libtorch_cuda.so)
frame #4: <unknown function> + 0x36ee950 (0x7f67274ee950 in /root/fcc/software/conda/envs/moe-infinity/lib/python3.10/site-packages/torch/lib/libtorch_cuda.so)
frame #5: at::_ops::mm::redispatch(c10::DispatchKeySet, at::Tensor const&, at::Tensor const&) + 0x6e (0x7f675ba40a0e in /root/fcc/software/conda/envs/moe-infinity/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #6: <unknown function> + 0x4fcb163 (0x7f675dfcb163 in /root/fcc/software/conda/envs/moe-infinity/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #7: <unknown function> + 0x4fcbcf3 (0x7f675dfcbcf3 in /root/fcc/software/conda/envs/moe-infinity/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #8: at::_ops::mm::call(at::Tensor const&, at::Tensor const&) + 0xc0 (0x7f675ba90340 in /root/fcc/software/conda/envs/moe-infinity/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #9: <unknown function> + 0x1dab8b0 (0x7f675adab8b0 in /root/fcc/software/conda/envs/moe-infinity/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #10: at::native::matmul(at::Tensor const&, at::Tensor const&) + 0x49 (0x7f675adb4339 in /root/fcc/software/conda/envs/moe-infinity/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #11: <unknown function> + 0x3081f90 (0x7f675c081f90 in /root/fcc/software/conda/envs/moe-infinity/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #12: at::_ops::matmul::call(at::Tensor const&, at::Tensor const&) + 0xc0 (0x7f675bbcb750 in /root/fcc/software/conda/envs/moe-infinity/lib/python3.10/site-packages/torch/lib/libtorch_cpu.so)
frame #13: MixtralMoEDenseActDense::forward(at::Tensor) + 0x66 (0x7f667ccbdcf6 in /root/.cache/torch_extensions/py310_cu124/prefetch/prefetch.so)
frame #14: ExpertDispatcher::GPUExecFunc(int) + 0x2ec (0x7f667ccb419c in /root/.cache/torch_extensions/py310_cu124/prefetch/prefetch.so)
frame #15: base::detail::startThread(void*) + 0x8d (0x7f667ccce99d in /root/.cache/torch_extensions/py310_cu124/prefetch/prefetch.so)
frame #16: <unknown function> + 0x7ea5 (0x7f6780207ea5 in /lib64/libpthread.so.0)
frame #17: clone + 0x6d (0x7f677f2feb0d in /lib64/libc.so.6)
 - expert_dispatcher.cpp:381
已放弃

运行read_me.py遇到这个问题 没有更改代码 模型也是hugging上下载的

**Comment by drunkcoding** [comment]
Current support to GPTQ is depreciated, it will be back in the future release. We only support original version for now.
